lib/ex_l7/ack.ex

```diff
@@ -0,0 +1,155 @@
+defmodule ExL7.Ack do
+  @moduledoc """
+  Provides functions to generate HL7 acknowledgement messages
+  """
+
+  import ExL7.Message.Header
+
+  @doc """
+  Returns an HL7 acknowledgement message as a string
+  """
+  def acknowledge(%ExL7.Message{} = l7_message, sequence) do
+    get_ack_hl7(l7_message, sequence, "AA")
+  end
+
+  @doc """
+  Returns an HL7 negative acknowledgement message as a string with AE as the code.
+  """
+  def error(l7_message_or_ack_config, sequence, reason \\ "")
+
+  def error(%ExL7.Message{} = l7_message, sequence, reason) do
+    get_ack_hl7(l7_message, sequence, "AE", reason)
+  end
+
+  def error(%ExL7.Ack.Config{} = ack_config, sequence, reason) do
+    get_ack_hl7(ack_config, sequence, "AE", reason)
+  end
+
+  def default_error(sequence, reason \\ "") do
+    error(%ExL7.Ack.Config{}, sequence, reason)
+  end
+
+  @doc """
+  Returns an HL7 negative acknowledgement message as a string with AR as the code.
+  """
+  def reject(l7_message_or_ack_config, sequence, reason \\ "")
+
+  def reject(%ExL7.Message{} = l7_message, sequence, reason) do
+    get_ack_hl7(l7_message, sequence, "AR", reason)
+  end
+
+  def reject(%ExL7.Ack.Config{} = ack_config, sequence, reason) do
+    get_ack_hl7(ack_config, sequence, "AR", reason)
+  end
+
+  def default_reject(sequence, reason \\ "") do
+    reject(%ExL7.Ack.Config{}, sequence, reason)
+  end
+
+  @doc """
+  Returns an HL7 negative acknowledgement message as a string with a custom code
+  """
+  def other(l7_message_or_ack_config, sequence, code, reason \\ "")
```

Comment:
The description is actually the problem. `other` is intended to let the caller specify the code + reason, instead of the code being chosen for them (`reject` chooses the code AR, `error` chooses AE, etc). 

We don't specifically use `other` right now, but this is in case the client expects a specific code (idk, like "AZ?") for a specific use case (maybe timezone is off, so I say AZ? I'm just making this up).